content/en/real_user_monitoring/guide/browser-sdk-upgrade.md

```diff
@@ -10,7 +10,57 @@ further_reading:
   text: "Use Datadog Session Replay to view real-time user journeys"
 ---
 
-## Overview
+This guide list all the migration steps between the different major versions of the Browser SDK.
+
+## From v3 to v4
+
+Several breaking changes were made to RUM and logs Browser SDK with the v4 version.
+
+## Changes
+
+### Intake URLs
+
+We changed URLs of where the Browser SDK data is sent. Please make sure that your [Content Security Policy is up to date][6].
+
+### Minimal Typescript version support
+
+The Browser SDK is now incompatible with TypeScript below v3.8.2. If you are using TypeScript, please make sure that the version you are using is greater or equal to v3.8.2.
+
+### Tags syntax
+
+version, env and service initialization parameters are sent as tags to Datadog. The Browser SDK is now slightly sanitizes them (to ensure that they don't unexpectedly generate multiple tags) and prints a warning if those values don't meet the tag requirements syntax.
+
+### Stricter initialization parameters typings
+
+TypeScript types representing initialization parameters are now stricter and may reject previously accepted unsupported parameters. If you get typechecking issues, please make sure you are only providing supported initialization parameters.
+
+### Privacy options precedence
+
+When multiple privacy options are specified on the same element, we now apply the most restrictive one to avoid unexpectedly leaking privacy (ex: if both `dd-privacy-allow` and `dd-privacy-hidden` classes are specified on the same element, we now consider it hidden instead of allow).
+
+### Action names computation
+
+When computing automatic actions target name on elements without `data-dd-action-name` attribute, we are using the "inner text" of this element. If it contains some elements that specify `data-dd-action-name`, their text is now removed inside the ancestor inner text.
```

Comment:
Struggling a bit with this one. Could we instead open with identifying the change the person might see? Something like:

```suggestion
Some automatic action names (ones without a `data-dd-action-name` attribute) might change within Datadog App because they are generated using the _inner text_ of this element. If the action contains some elements that specify `data-dd-action-name`, their text is removed inside the ancestor inner text.
```
Is the last sentence necessary? I replaced the word "it" with "the action" but I wasn't actually sure if that's correct. Do we really need to be this specific about how the action name will be different now? If so, can we be clearer (maybe provide an example)?

Reply:
I have reworked this part, let me know what you think

Reply:
Now I get it. I gave it a grammar rewrite, and committed. Are you ready to publish this?

Reply:
All good, thanks a lot for your review!